src/datasets/arrow_dataset.py

```diff
@@ -3806,33 +3811,36 @@ def _select_with_indices_mapping(
     @fingerprint_transform(inplace=False, ignore_kwargs=["load_from_cache_file", "indices_cache_file_name"])
     def sort(
         self,
-        column: str,
-        reverse: bool = False,
-        kind: str = None,
-        null_placement: str = "last",
+        column_names: Union[str, Sequence_[str]],
+        reverse: Union[bool, Sequence_[bool]] = False,
+        kind="deprecated",
+        null_placement: str = "at_end",
         keep_in_memory: bool = False,
         load_from_cache_file: Optional[bool] = None,
         indices_cache_file_name: Optional[str] = None,
         writer_batch_size: Optional[int] = 1000,
         new_fingerprint: Optional[str] = None,
```

Comment:
Maybe keep `column=` as a deprecated keyword argument as well ?

Reply:
For `dataset.sort(column="col")` to keep working, we also need to define a default value for `column_names`, so not sure about this.

Reply:
Yes, but `column_names` is a **positional-only** argument, so it needs a default value, or `sort` would throw the `missing argument error` when one calls it as `dataset.sort(column="col")`. And this makes the code a bit messy IMO.

Reply:
I checked and couldn't find documentation or script passing `column=` so I think it's fine